docs/src/proposals/optimistic_confirmation.md

```diff
@@ -0,0 +1,380 @@
+                       |                      +---+---+
+                       |                          |
+                       |                          |
+                       |                          |
+                       |                      +---+---+
+                       |                      |       |
+                       |                      |       |  S.last
+                       |                      |       |
+                       |                      +-------+
+                       |
+                   +---+---+
+                   |       |
+    B'(Finalized)  |       |
+                   |       |
+                   +-------+
+
+
+`Proof`: Let `Vote(X, S)` be a vote in the `Delinquent Optimistic Votes` set,
+where the for the "optimistcally confirmed" block `B`, `X <= B <= S.last`.
+
+Because `X` is a parent of `B`, and `B'` is not a parent or ancestor of `B`,
+then:
+
+* `B' != X`
+* `B'` is not a parent of `X`
+
+Now consider if `B'` < `X`:
+
+`Case B' < X`: We wll show this is a violation of lockouts. 
+From above, we know `B'` is not a parent of `X`. Then because `B'` was rooted,
+and `B'` is not a parent of `X`, then the validator should not have been able
+to vote on the higher slot `X` that does not descend from `B'`.
+
+### Proof of Safety:
+We now aim to show at least one of the validators in the `Delinquent` set
+violated a slashing rule.
+
+By definition, in order to root `B'`, each valdator `V` in `Delinquent`
+must have each made some "switching vote" of the form `Vote(X_v, S_v)` where:
+
+* `S_v.last > B'`
+* `S_v.last` is a descendant of `B'`, so it can't be a descendant of `B`
+* Because `S_v.last` is not a descendant of `B`, then `X_v` cannot be a
+descendant or ancestor of `B`.
+
+Recall also this delinquent validator `V` must also have made some vote
+`Vote(X, S)` in the `Delinquent Optimistic Votes` set. By definition of that
+set (all votes optimistically confirmed `B`), we know `S.last >= B >= X`
+
+By `Lemma 2` we know `B' > X`, and from above `S_v.last > B'`, so then
+`S_v.last > X`. Because `X_v != X` (cannot be a descendant or ancestor of 
+`B` from above), then by the slashing rules then, we know `X_v > S.last`.
+From above, `S.last >= B >= X` so for all such "switching votes", `X_v > B`.
+
+Ordering all these "switching votes" in time, let `V` to be the validator
+that first submitted such a "swtching vote" `Vote(X', S')`, where from the
+properties derived from above, `X' > B`
+
+Now let `Vote(X, S)` be the vote in `Delinquent Optimistic Votes` made by
+validator `V`. 
+
+Given `Vote(X, S)` and the the "Optimistic Slashing" rules, because
+`X' > B > X`, then `X' > S.last`.
+
+In order to perform such a "switching vote" to `X'`, a switching proof
+`SP(Vote(X, S), Vote(X', S')` must show `> 1/3` of stake being locked
+out at this validator's latest vote, `S.last`. Combine this `>1/3` with the
+fact that the set of validators in `Delinquent` consists of `> 2/3` of the
```

Comment:
I thought `Delinquent` represented the 1/3 of stake that both rooted B' and was included in the optimistic confirmation of B?

Should this be: "that the set of validators in `Delinquent`" -> "the set of validators who rooted `B'`"?

Reply:
@jstarry you're totally right this part was completely unclear. I've reworked it in my latest commit, what do you think?

Reply:
Makes more sense now! Left some comments to make sure I'm understanding things correctly